suite-native/accounts/src/selectors.ts

```diff
@@ -81,23 +84,54 @@ export const selectIsAccountAlreadyDiscovered = (
         ),
     );
 
-export const selectAccountFiatBalance = (state: NativeAccountsRootState, accountKey: string) => {
-    const fiatRates = selectCurrentFiatRates(state);
-    const account = selectAccountByKey(state, accountKey);
-    const localCurrency = selectFiatCurrencyCode(state);
+export const selectAccountFiatBalance = createMemoizedSelector(
+    [
+        selectCurrentFiatRates,
+        selectAccountByKey,
+        selectFiatCurrencyCode,
+        (_, _accountKey: AccountKey, shouldIncludeStaking?: boolean) =>
+            shouldIncludeStaking ?? true,
+        (
+            _,
+            _accountKey: AccountKey,
+            _shouldIncludeStaking?: boolean,
+            shouldIncludeTokens?: boolean,
+        ) => shouldIncludeTokens ?? true,
+    ],
+    (fiatRates, account, localCurrency, shouldIncludeStaking, shouldIncludeTokens) => {
+        if (!account) {
+            return '0';
+        }
 
-    if (!account) {
-        return '0';
-    }
+        const totalBalance = getAccountFiatBalance({
+            account,
+            rates: fiatRates,
+            localCurrency,
+            shouldIncludeStaking,
+            shouldIncludeTokens,
+        });
 
-    const totalBalance = getAccountFiatBalance({
-        account,
-        rates: fiatRates,
-        localCurrency,
-    });
+        if (!totalBalance) {
+            return '0';
+        }
 
-    return totalBalance;
-};
+        return totalBalance;
+    },
+);
+
+export const selectAccountTokenFiatBalance = createMemoizedSelector(
+    [selectCurrentFiatRates, selectFiatCurrencyCode, selectAccountByKey, selectAccountTokenInfo],
+    (fiatRates, localCurrency, account, tokenInfo) => {
+        if (!account || !fiatRates || !tokenInfo) return '0';
+        const { contract, balance } = tokenInfo;
+        const fiatRateKey = getFiatRateKey(account.symbol, localCurrency, contract);
+        const rate = fiatRates[fiatRateKey]?.rate;
+
+        if (!rate || !balance) return '0';
+
+        return toFiatCurrency(balance, rate) ?? '0';
```

Comment:
Where is the `decimals` value reflected in the calculation?

Reply:
hopefully fixed